src/sqlfluff/dialects/dialect_teradata.py

```diff
@@ -778,6 +779,22 @@ class UnorderedSelectStatementSegment(ansi.UnorderedSelectStatementSegment):
     )
 
 
+class DeleteStatementSegment(BaseSegment):
+    """A `DELETE` statement.
+
+    DEL[ETE] FROM <table name> [ WHERE <search condition> ]
+    """
+
+    type = "delete_statement"
+    # match grammar. This one makes sense in the context of knowing that it's
+    # definitely a statement, we just don't know what type yet.
+    match_grammar: Matchable = Sequence(
+        OneOf("DELETE", "DEL"),
```

Comment:
Thanks, though I feel it's better to put `OneOf("DELETE", "DEL"),` in `dialect_ansi` itself instead of in `dialect_teradata`.

https://github.com/sqlfluff/sqlfluff/blob/ba1e6c9e7623125191f050586bdf4d86a693f698/src/sqlfluff/dialects/dialect_ansi.py#L3504

Reply:
I usually slightly prefer to put non-ANSI stuff there still instead of copying the whole function again in another dialect, but I guess I'm indifferent. Will leave to @WittierDinosaur if he is ok with the current code.